lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -258,14 +253,135 @@ def _prepare_to_run(self, params, concierge_params=None) -> JobSubmissionParamet
+        )
+        return self._generate_job_submission_params(job_id, params)
+
+    def _run_multiple(self, runjob_params):
+        """
+        Get the job records, bulk save them, then submit to condor.
+        If any condor submission fails, abort all of the jobs
+        :return:
+        """
+        # Save records to db
+        job_records = []
+        for runjob_param in runjob_params:
+            job_records.append(
+                self._init_job_rec(self.sdkmr.get_user_id(), runjob_param, save=False)
+            )
+        job_ids = self.sdkmr.save_jobs(job_records)
+
+        # Generate job submission params
+        job_submission_params = []
+        for i, job_id in enumerate(job_ids):
+            job_submission_params.append(
+                self._generate_job_submission_params(job_id, runjob_params[i])
+            )
+            assert job_id == job_submission_params[i].job_id
+
+        # Takes 2.5200018882751465 for 100 records, can shave off 2.5 secs by making this async
+        for job_id in job_ids:
+            self.sdkmr.get_kafka_client().send_kafka_message(
+                message=KafkaCreateJob(
+                    job_id=str(job_id), user=self.sdkmr.get_user_id()
+                )
+            )
+        # Submit to Condor
+        try:
+            submission_ids = self._submit_multiple(job_submission_params)
+            return submission_ids
+        except Exception as e:
+            self._abort_multiple_jobs(job_ids)
+            raise e
+
+    def _update_to_queued_multiple(self, job_ids, scheduler_ids):
+        """
+        This is called during job submission. If a job is terminated during job submission,
+        we have the chance to re-issue a termination and remove the job from the Job Queue
+        """
+        if len(job_ids) != len(scheduler_ids):
+            raise Exception(
+                "Need to provide the same amount of job ids and scheduler_ids"
+            )
+        jobs_to_update = list(map(JobIdPair, job_ids, scheduler_ids))
+        self.sdkmr.get_mongo_util().update_jobs_to_queued(jobs_to_update)
+        jobs = self.sdkmr.get_mongo_util().get_jobs(job_ids)
+
+        for job in jobs:
+            job_id = str(job.id)
+            if job.status == Status.queued.value:
+                self.sdkmr.get_kafka_client().send_kafka_message(
+                    message=KafkaQueueChange(
+                        job_id=job_id,
+                        new_status=Status.queued.value,
+                        previous_status=Status.created.value,  # TODO maybe change this to allow for estimating jobs
+                        scheduler_id=job.scheduler_id,
+                    )
+                )
+            elif job.status == Status.terminated.value:
+                # Remove from the queue, now that the scheduler_id is available
+                self._safe_cancel(job_id, TerminatedCode.terminated_by_user)
```

Comment:
What if the job was terminated for some other reason?

Reply:
Nothing actually happens since
```
    def cancel_job(self, job_id=None, terminated_code=None):
        with self.mongo_engine_connection():
            j = self.get_job(job_id)
            if self.check_if_already_finished(j.status):
                return False
```